### DISCUSSION TITLE
How to add domains via API?

### DISCUSSION THREAD

**Discussion by sharrc** [postalserver/postal#1462] [Help with using Postal]
Hello, I have a basic setup with postal on Ubuntu 18.04.

And I have to allow users to be able to add a domain using an API and get a DKIM public key back in response which they can they add into their DNS records.

Is there a way this is possible right now? Or do I have to use alternatives like a mix of Postfix + Amavis + Dovecot, or something on those lines...?

My requirements include:
1. Allowing multiple domains to be added anytime
2. Only sending emails from any user on that domain (No receiving)
3. Being able to customize the DNS records like the DKIM's selector, which looks something like: postal_Nkj24._domainkey.example.com
Would like to replace the "postal" bit here...

There is a commit actually that comes close to what I want: https://github.com/postalhq/postal/commit/2dd44e894e6522e485d91e9cd404f76959173af3

But this is not merged into the latest master version, and also there's no documentation for it.

Please help! Thanks!

Tagging: @willpower232 @adamcooke

**Comment by willpower232** [comment]
Also in the list of PRs is https://github.com/postalhq/postal/pull/992 which sounds much closer to what you want.

These feature requests have not been merged as there are other priorities and no guarantee they will be merged. If you want to make use of that code, you wlil have to fork Postal and merge it yourself to create the version of Postal you desire but you would have to figure out how to use it on your own.

Edit: fixed link

**Reply by PrabhuShanmughapriyan** [reply]
@willpower232  can we please merge this #992 it will be really useful for us

**Reply by violuke** [reply]
We would love to switch to Postal, but would need this to be able to do so. If there's any appetite to merge this, that would be great for us. Thank you :pray:

Edit: I've just seen https://github.com/postalserver/postal/pull/992#issuecomment-2047297129, that's great news!

**Comment by charlesmudy** [comment]
Well, several years later we still don't have this, yet @adamcooke @willpower232 someone that created a merge solution was rejected.

**Reply by csiocc** [reply]
[Postal-api-extended](https://github.com/csiocc/postal-api-extended) will be maintained until we get a V2 api

**Comment by uhlhosting** [comment]
Yes this is something very useful to have, petty the last PR was closed in 2024 and 2 years only... later still no changes, no updates... No direction! Pitty such a potent product, yet would require more involvement to at least allow community to work on adding value, not this approach. @adamcooke any idea when we will be close to test a new api?

**Reply by adamcooke** [reply]
> Yes this is something very useful to have, petty the last PR was closed in 2024 and 2 years only... later still no changes, no updates... No direction! Pitty such a potent product, yet would require more involvement to at least allow community to work on adding value, not this approach. @adamcooke any idea when we will be close to test a new api?

I'd say within the next 3 months. I have a need for it myself so will be able to find some time for it.

**Reply by adamcooke** [reply]
I've posted an update [here](https://github.com/orgs/postalserver/discussions/1977#discussioncomment-17764756) about progress on API v2.

**Reply by uhlhosting** [reply]
@adamcooke looking forward for this to get out, like a child for a fresh baked cookie.
